File: java/src/test/java/problems101_200/BestTimeToBuyAndSellStock2Test.java
package problems101_200;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class BestTimeToBuyAndSellStock2Test extends SampledTest {
    @Override
    protected String getSampleFilename() {
        return "best-time-to-buy-and-sell-stock-ii.json";
    }

    @Test
    public void test() {
        for (BestTimeToBuyAndSellStock2 solution : BestTimeToBuyAndSellStock2.values()) {
            for (BestTimeToBuyAndSellStock2Sample sample
                : getSamples(BestTimeToBuyAndSellStock2Sample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.maxProfit(sample.input.prices))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/BestTimeToBuyAndSellStock2Test.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class BestTimeToBuyAndSellStock2Test : SampledTest() {
    override fun getSampleFilename(): String = "best-time-to-buy-and-sell-stock-ii.json"

    @Test
    fun test() =
        BestTimeToBuyAndSellStock2.entries.forEach { solution ->
            getSamples(Array<BestTimeToBuyAndSellStock2Sample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.maxProfit(input.prices))
                        .isEqualTo(output)
                }
        }
}
